### ISSUE TITLE
Gap in Snowflake `CREATE USER ... RSA_PUBLIC_KEY = '...'`

### ISSUE THREAD

**Issue by juli4nb4dillo** [sqlfluff/sqlfluff#6712]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When parsing:

<https://docs.snowflake.com/en/sql-reference/sql/create-user>

```sql
CREATE USER username
RSA_PUBLIC_KEY = '<BASE 64 ENCODED PUBLIC KEY>';
```

SQLFluff throws a "Found unparsable section" error.

### Expected Behaviour

This should parse ok.

### Observed Behaviour

```sh
L:  47 | P:  22 |  PRS | Line 47, Position 22: Found unparsable section: "WITH
                       | rsa_public_key\n
```

### How to reproduce

```sql
CREATE USER username
RSA_PUBLIC_KEY = '<BASE 64 ENCODED PUBLIC KEY>';
```

### Dialect

Snowflake

### Version

sqlfluff, version 3.3.1
Python 3.11.11

### Configuration

[sqlfluff]
exclude_rules = ambiguous.column_count,
                references.from
max_line_length = 120
[sqlfluff:templater:dbt]
project_dir = ./src/
[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower 
[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper
[sqlfluff:rules:layout.long_lines]
ignore_comment_clauses=True
ignore_comment_lines=True

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
